### PR TITLE
[pytest] improvements for pytest infrastructure

### PR DESCRIPTION
**Pull request by stepanblyschak**
### Description of PR

Summary:

- pytest:
  - W/A some issues with pytest-ansible plugin and provided fixtures
  - add ptf adapter package and fixture to send/receive PTF traffic controlled from pytest
  - translate fdb test to use PTF adapter
  - Divide LLDP test into two test cases
- ansible
  - add pytest_runner.yml to wrap pytest in ansible run
  - remove lldp.yml test code ansible and lldp_neighbor.yml. Instead lldp.yml will run pytest test_lldp.py

TODO: 
- [x] upgrade PTF version and install PTF in sonic-mgmt container
- [x] get community string for LLDP test from eos.yml or introduce another configuration file
    ( in our testbed community string is 'public' while test_lldp.py hardcodes 'strcommunity' )

Fixes # (issue)

### Type of change

- [] Bug fix
- [x] Testbed and Framework(new/improvement)
- [] Test case(new/improvement)

#### How did you verify/test it?

```
stepanb@62c67c04c2bd:/stepanb/sonic-mgmt/tests$ py.test -k 'not platform'  --testbed=$SWITCH-t0 --testbed_file=testbed.csv --inventory=../ansible/inventory --host-pattern $SWITCH-t0 -v --ignore=ptftests
=========================================================================================================== test session starts ===========================================================================================================
platform linux2 -- Python 2.7.12, pytest-4.6.4, py-1.8.0, pluggy-0.12.0 -- /usr/bin/python
cachedir: .pytest_cache
Using --randomly-seed=1564067815
ansible: 2.0.0.2
rootdir: /stepanb/sonic-mgmt/tests, inifile: pytest.ini
plugins: randomly-1.2.3, ansible-2.0.2
collected 16 items / 11 deselected / 5 selected                                                                                                                                                                                           

test_bgp_fact.py::test_bgp_facts PASSED                                                                                                                                                                                             [ 20%]
test_lldp.py::test_lldp PASSED                                                                                                                                                                                                      [ 40%]
test_lldp.py::test_lldp_neighbor PASSED                                                                                                                                                                                             [ 60%]
test_bgp_speaker.py::test_bgp_speaker SKIPPED                                                                                                                                                                                       [ 80%]
fdb/test_fdb.py::test_fdb PASSED                                                                                                                                                                                                    [100%]

========================================================================================== 4 passed, 1 skipped, 11 deselected in 147.06 seconds ===========================================================================================
```

Verify ansible LLDP test

```
ansible-playbook test_sonic.yml -i inventory -l r-boxer-sw01 -e testbed_name=r-boxer-sw01-t0 -e topo=t0 -e testcase_name=lldp -vvvvvv
```

### CODE REVIEW

File: ansible/roles/test/tasks/pytest_runner.yml [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+  when: test_mark is defined
+
+- name: append testbed name
+  set_fact:
+    pytest_cmd: '{{ pytest_cmd }} --testbed={{ testbed_name }}'
+
+# We could use ansible_inventory_sources magic variable but it is available in ansible >=2.5
+- name: append inventory file
+  set_fact:
+    pytest_cmd: '{{ pytest_cmd }} --inventory=../ansible/inventory'
```
Comment (lguohan) [inline ansible/roles/test/tasks/pytest_runner.yml]: can we get the inventory file name form the user input, not always this inventory file.

File: ansible/roles/test/tasks/pytest_runner.yml [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+
+- name: append testbed file
+  set_fact:
+    pytest_cmd: '{{ pytest_cmd }} --testbed_file=../ansible/testbed.csv'
```
Comment (lguohan) [inline ansible/roles/test/tasks/pytest_runner.yml]: can we get the testbed csv file name form the user input, not always this inventory file.

**Comment by lguohan** [comment]
LGTM overall

File: ansible/roles/test/tasks/pytest_runner.yml (hunk 2 above)
Comment (stepanblyschak) [reply ansible/roles/test/tasks/pytest_runner.yml]: For testbed.csv yes. I see some playbooks accept 'testbed_file' external variable I can use that as well.

File: ansible/roles/test/tasks/pytest_runner.yml (hunk 1 above)
Comment (stepanblyschak) [reply ansible/roles/test/tasks/pytest_runner.yml]: For inventory file I don't know. I found 'ansible_inventory_sources' magic variable which is in ansbile >2.5 only.
But I can accept 'inventory_file' as external variable from user.

Comment (stepanblyschak) [reply ansible/roles/test/tasks/pytest_runner.yml]: I found that 'inventory_file' is set to inventory passed by '-i' option. I'll use that

**Comment by dawnbeauty** [comment]
@stepanblyschak 
1. Can we support configuring nn_rcv_buff and nn_snd_buff size within ptf_nn_agent.conf.ptf.j2. Without that, may drop pkts due to heavy traffic on ptf side.
2. I think we should add stop_pcap when kill dataplane.

**Comment by stepanblyschak** [comment]
@dawnbeauty 
I believe PTF adapter will become more configurable when more tests will use it, right now its first stage for simple traffic tests like fdb, fib, acl which do not expect to send, receive much traffic.
1. Yes, we can and I believe we'll add this for copp and other tests which require to send lots of packets
2. We haven't started pcap, so we didn't stop pcap in kill(). I believe we can add pcap writer.

I'm about to convert ACL and everflow tests so I will consider your comments. Thanks

**Pull request merged**